projects/plugins/jetpack/extensions/blocks/nextdoor/nextdoor.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Nextdoor Block.
+ *
+ * @since $$next-version$$
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Nextdoor;
+
+use Automattic\Jetpack\Blocks;
+use Jetpack_Gutenberg;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	Blocks::jetpack_register_block(
+		__DIR__,
+		array( 'render_callback' => __NAMESPACE__ . '\load_assets' )
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Nextdoor block registration/dependency declaration.
+ *
+ * @param array $attr    Array containing the Nextdoor block attributes.
+ * @return string
+ */
+function load_assets( $attr ) {
+	if ( ! isset( $attr['url'] ) ) {
+		return;
+	}
+
+	$url = Jetpack_Gutenberg::validate_block_embed_url(
+		$attr['url'],
+		array( '/^http[s]?:\/\/((?:www\.)?nextdoor(?:.*)?\/(?:embed)\/[a-z0-9\/\?=_\-\.\,&%$#\@\!\+]*)/i' ),
```

Comment:
A simplified version of this regext would be to allow any non-whitespace character at the end of the URL
```suggestion
		array( '/^http[s]?:\/\/((?:www\.)?nextdoor(?:.*)?\/(?:embed)\/\S*)/i' ),
```